### PR TITLE
case insensitive login

### PR DESCRIPTION
**Pull request by harminius**
Make login case insensitive

authenticate() method is used in login through app as well as public login

migration tested on dev clone and went smoothly

### CODE REVIEW

File: server/mergin/auth/models.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ class User(db.Model):
         default=datetime.datetime.utcnow,
     )
 
+    __table_args__ = (
+        db.Index("ix_username_insensitive_unique", func.lower(username), unique=True),
```
Comment (varmar05) [inline server/mergin/auth/models.py:36]: we can probably remove original unique indexes?

File: server/migrations/community/07f2185e2428_case_insensitive_unique_idx.py [hunk 2]
```diff
@@ -25,9 +25,15 @@ def upgrade():
     conn.execute(
         "CREATE UNIQUE INDEX ix_email_insensitive_unique ON public.user (LOWER(email));"
     )
+    conn.execute("ALTER TABLE public.user DROP CONSTRAINT uq_user_email;")
```
Comment (varmar05) [inline server/migrations/community/07f2185e2428_case_insensitive_unique_idx.py:28]: we should follow the same naming convention with new constraints

**Review by varmar05 — APPROVED** [review]

File: server/migrations/community/07f2185e2428_case_insensitive_unique_idx.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "07f2185e2428"
+down_revision = "0e3fc92aeaaa"
```
Comment (MarcelGeo) [inline server/migrations/community/07f2185e2428_case_insensitive_unique_idx.py:15]: Check if ook

Comment (harminius) [reply server/migrations/community/07f2185e2428_case_insensitive_unique_idx.py]: updated, thanks

**Pull request merged**